src/pipeline_migration/actions/mod_task.py

```diff
@@ -0,0 +1,155 @@
+        metavar="PARAM-VALUE"
+    )
+    subparser_add_param.set_defaults(action=action_add_param)
+
+
+
+class ModTaskAddParamOperation(PipelineFileOperation):
+    def __init__(
+        self,
+        task_name: str,
+        param_name: str,
+        param_value: str,  # TODO: array, object, string values
+        git_add: bool = False,  # TODO
+    ) -> None:
+        self.task_name = task_name
+        self.param_name = param_name
+        self.param_value = param_value
+        self.git_add = git_add
+
+    def handle_pipeline_file(self, file_path: FilePath, loaded_doc: Any, style: YAMLStyle) -> None:
+        tasks = loaded_doc["spec"]["tasks"]
+        if self._add_param(tasks, str(file_path)):
+            pass
+            #if self.git_add:
+            #    git_add(file_path)
+            #    logger.info("%s is added to git index.", file_path)
+
+    def handle_pipeline_run_file(
+        self, file_path: FilePath, loaded_doc: Any, style: YAMLStyle
+    ) -> None:
+        tasks = loaded_doc["spec"]["pipelineSpec"]["tasks"]
+        if self._add_param(tasks, str(file_path)):
+            pass
+            #if self.git_add:
+            #    git_add(file_path)
+            #    logger.info("%s is added to git index.", file_path)
+
+    def _add_param(self, tasks: CommentedSeq, pipeline_file: str) -> bool:
+        # TODO: add new params when params parent doesn't exist
+        # TODO: add new param when name doesn't exist
+        for task in tasks:
+            if task["name"] == self.task_name:
+                for param in task.get("params", []):
+                    if param['name'] == self.param_name:
+                        if param['value'] != self.param_value:
+                            yaml = YAML()
+                            stream = StringIO()
+                            yaml.dump(
+                                [{"name": self.param_name, "value": self.param_value}],
+                                stream=stream,
+                            )
+                            yaml_output = stream.getvalue()
+                            lc = param.lc
+
+                            # Indent each line of the YAML output by the column position
+                            indented_lines = []
+                            for line in yaml_output.splitlines():
+                                if line.strip():  # Only indent non-empty lines
+                                    indented_lines.append(" " * (lc.col - 2) + line)  # list indentation adds extra 2 spaces
+                                else:
+                                    indented_lines.append(line)
+                            new_val = "\n".join(indented_lines)
+                            # TODO: this works by random magic only, it only assumes that  output will be exactly 2 lines
+                            # lc exists only in node, there is no lc in dict keys, thus we don't know exact structure
+                            # more heuristics is needed
```

Comment:
I'm not sure I get your point here, especially for the `exact structure`. Maybe `YAMLStyle.detect` and `BlockSequenceIndentation.is_consistent are helpful.

For this case of adding a task parameter, I think we don't need to care about the `lc` of dict keys. By taking this example:

```yaml
tasks:
- name: clone
  params:
  - name: git-url
  - name: revision
- name: build
```

 as mentioned above `tasks[0]["params"]` is a `CommentedSeq` object, for adding parameter `revision`, its indentation is the `tasks[0]["params"].lc.col`.